---
title: "Analysis tools and QC"
project:
  execute-dir: project
editor_options: 
  chunk_output_type: console
---

::: {.callout-note}
## Learning outcomes

**After having completed this chapter you will be able to:**

- Load single cell data into R
- Explain the basic structure of a `Seurat` object and extract count data and metadata
- Calculate and visualize quality measures based on:
    - mitochondrial genes
    - ribosomal genes
    - hemoglobin genes
    - relative gene expression
- Interpret the above quality measures per cell.
- Perform cell filtering based on quality thresholds
:::

## Material

{{< downloadthis ../assets/pdf/04_analysis_tools_qc.pdf dname="04_analysis_tools_qc" label="Download the presentation" icon="filetype-pdf" >}}

{{< video https://youtu.be/YT7xeHVnTw4?si=YkNt4QOpgqMrLulb >}}

## Exercises

### Loading scRNAseq data

The next step after the generation of the count matrices with `cellranger count`, is the data analysis. The `R` package `Seurat` is currently the most popular software to do this. To start working with `Seurat` you can load it into your environment like this:
