---
title: "Exercise 3"
editor: source
editor_options: 
  chunk_output_type: console
---

# ChIP and RNA sequencing data plots

## Learning Objectives


## Load Libraries

```{r}
#| warning: false
#| message: false
library(SummarizedExperiment)
library(EnrichedHeatmap)
library(gUtils)
library(rtracklayer)
library(circlize)
```


## ChIP-seq SE object
Here we load the `SummarizedExperiment` for the ChIP-seq data
```{r}
h3k4me3 <- readRDS("data/h3k4me3_se.rds")
h3k4me1 <- readRDS("data/h3k4me1_se.rds")
h3k27me3 <- readRDS("data/h3k27me3_se.rds")
h3k27ac <- readRDS("data/h3k27ac_se.rds")
```

## Question 1

**Similar to ATAC-seq data, please create `EnrichedHeatmap` for all ChIP-seq datasets and observe difference in genome activity at different genomic regions.**


## RNA-seq data
```{r}
rna <- readRDS("data/rna_se.rds")
```

## Heatmap of counts
For RNA-seq data, we use `ComplexHeatmap` package to make heatmaps

```{r}
Heatmap(matrix = assay(rna, "logCPM"), 
        name = "RNA", 
        cluster_columns = FALSE, 
        show_row_names = FALSE
)
```

### Changing visualization to fold change

```{r}
counts <- assay(rna, "logCPM") - rowMeans(
  assay(rna, "logCPM")[, grep(pattern = "11half", x = colnames(assay(rna, "logCPM")),
                              value = T)]
)
```

```{r}
Heatmap(matrix = counts, 
        name = "RNA", 
        cluster_columns = FALSE, 
        show_row_names = FALSE
)
```

It is now more clear, which genes are up-regulated and which are down-regulated.

### Adding colum annotations
```{r}
colData(rna)
```

We can add groups information to the heatmap
```{r}
Heatmap(matrix = counts, 
        name = "RNA", 
        cluster_columns = FALSE, 
        show_row_names = FALSE, 
        show_column_names = FALSE,
        top_annotation = HeatmapAnnotation(df = colData(rna)[,2,drop = FALSE]),
        width = unit(4, "cm"),
        height = unit(8, "cm"),
        heatmap_legend_param = list(
            at = scales::pretty_breaks(n = 3)(range(counts, na.rm = TRUE)), 
            title = "RNA"
        )
)
```

## Question 2

**Similar to EnrichedHeatmap, we can also split RNA-seq data based on logFC. Can you try to do this?**

:::{.callout-tip collapse="true"}
```{r}
logFC <- rowData(rna)[,"logFC"]
names(logFC) <- rownames(rowData(rna))

logFC <- ifelse(logFC > 0, "Up regulated", "Down regulated")

Heatmap(matrix = counts, 
        name = "RNA", 
        cluster_columns = FALSE, 
        show_row_names = FALSE, 
        show_column_names = FALSE,
        top_annotation = HeatmapAnnotation(df = colData(rna)[,2,drop = FALSE]),
        width = unit(4, "cm"),
        height = unit(8, "cm"),
        heatmap_legend_param = list(
            at = scales::pretty_breaks(n = 3)(range(counts, na.rm = TRUE)), 
            title = "RNA"
        ),
        row_split = logFC
)
```
:::

:::{.callout-important}
**`Heatmap` and `EnrichedHeatmap` have a lot of options common.**
:::
