<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.31">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Exercise 7 â€“ Biology-informed Multiomics Data Integration and Visualization</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../assets/SIB_logo.svg" rel="icon" type="image/svg+xml">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-e1a5c8363afafaef2c763b6775fbf3ca.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-ab62ab6abcf2bb06eba42c0c53629186.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Source+Sans+Pro">
<!-- Matomo --><script>
  var _paq = window._paq = window._paq || [];
  /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="https://matomo.sib.swiss/";
    _paq.push(['setTrackerUrl', u+'matomo.php']);
    _paq.push(['setSiteId', '220']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
  })();
</script><!-- End Matomo Code -->
</head>
<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Biology-informed Multiomics Data Integration and Visualization</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../precourse_preparations.html"> 
<span class="menu-text">Precourse preparations</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../course_schedule.html"> 
<span class="menu-text">Course schedule</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-day-1" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">day 1</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-day-1">
<li>
    <a class="dropdown-item" href="../day1/day1-1_dataset.html">
 <span class="dropdown-text">Exercise 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day1/day1-2_atac.html">
 <span class="dropdown-text">Exercise 2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day1/day1-3_chip_rna.html">
 <span class="dropdown-text">Exercise 3</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-day-2" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">day 2</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-day-2">
<li>
    <a class="dropdown-item" href="../day2/day2-1_overlap_matrix.html">
 <span class="dropdown-text">Exercise 4</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day2/day2-2_atac_rna.html">
 <span class="dropdown-text">Exercise 5</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day2/day2-3_multiomics.html">
 <span class="dropdown-text">Exercise 6</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day2/day2-4_enrichment_analysis.html">
 <span class="dropdown-text">Exercise 7</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sib-swiss/biology-informed-multiomics-training"> <i class="bi bi-github" role="img" aria-label="GitHub">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Exercise 7</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../assets/SIB_LogoQ_GBv.svg" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      </div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li>
<a href="#enrichment-analysis" id="toc-enrichment-analysis" class="nav-link active" data-scroll-target="#enrichment-analysis">Enrichment analysis</a>
  <ul class="collapse">
<li><a href="#load-libraries" id="toc-load-libraries" class="nav-link" data-scroll-target="#load-libraries">Load Libraries</a></li>
  <li><a href="#overlapmatrix" id="toc-overlapmatrix" class="nav-link" data-scroll-target="#overlapmatrix">OverlapMatrix</a></li>
  <li>
<a href="#running-enrichment-analysis" id="toc-running-enrichment-analysis" class="nav-link" data-scroll-target="#running-enrichment-analysis">Running enrichment analysis</a>
  <ul class="collapse">
<li><a href="#output" id="toc-output" class="nav-link" data-scroll-target="#output">Output</a></li>
  <li><a href="#interpretation" id="toc-interpretation" class="nav-link" data-scroll-target="#interpretation">Interpretation</a></li>
  </ul>
</li>
  <li><a href="#question" id="toc-question" class="nav-link" data-scroll-target="#question">Question</a></li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<h1 class="title">Exercise 7</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><section id="enrichment-analysis" class="level1"><h1>Enrichment analysis</h1>
<section id="load-libraries" class="level2"><h2 class="anchored" data-anchor-id="load-libraries">Load Libraries</h2>
<div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/rGREAT">rGREAT</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Mmusculus.UCSC.mm10.knownGene</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Mm.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/simplifyEnrichment">simplifyEnrichment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">flexclust</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="overlapmatrix" class="level2"><h2 class="anchored" data-anchor-id="overlapmatrix">OverlapMatrix</h2>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">overlap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"output/overlap_anno2.rds"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="running-enrichment-analysis" class="level2"><h2 class="anchored" data-anchor-id="running-enrichment-analysis">Running enrichment analysis</h2>
<div class="cell">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">gr</span> <span class="op">&lt;-</span> <span class="va">overlap</span><span class="op">[</span><span class="va">overlap</span><span class="op">$</span><span class="va">ATAC_RNA</span> <span class="op">==</span> <span class="st">"Active Promoters"</span><span class="op">]</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"BP"</span>, <span class="st">"mm10"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>* TSS source: TxDb.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* check whether TxDb package 'TxDb.Mmusculus.UCSC.mm10.knownGene' is installed.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* gene ID type in the extended TSS is 'Entrez Gene ID'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* restrict chromosomes to 'chr1, chr2, chr3, chr4, chr5, chr6, chr7, chr8, chr9, chr10,
    chr11, chr12, chr13, chr14, chr15, chr16, chr17, chr18, chr19, chrX, chrY, chrM'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* 20593/24515 protein-coding genes left.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* update seqinfo to the selected chromosomes.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* TSS extension mode is 'basalPlusExt'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* construct the basal domains by extending 5000bp to upstream and 1000bp to downsteram of TSS.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* calculate distances to neighbour regions.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* extend to both sides until reaching the neighbour genes or to the maximal extension.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* use GO:BP ontology with 15445 gene sets (source: org.Mm.eg.db).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* check gene ID type in `gene_sets` and in `extended_tss`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* use whole genome as background.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* remove excluded regions from background.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* overlap `gr` to background regions (based on midpoint).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* in total 99 `gr`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* overlap extended TSS to background regions.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* check which genes are in the gene sets.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* only take gene sets with size &gt;= 5.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* in total 9412 gene sets.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* overlap `gr` to every extended TSS.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>* perform binomial test for each biological term.</code></pre>
</div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/plotVolcano-dispatch.html">plotVolcano</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="day2-4_enrichment_analysis_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/plotRegionGeneAssociations-dispatch.html">plotRegionGeneAssociations</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="day2-4_enrichment_analysis_files/figure-html/unnamed-chunk-3-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getRegionGeneAssociations-dispatch.html">getRegionGeneAssociations</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 99 ranges and 112 metadata columns:
            seqnames              ranges strand |        ATAC_annotation
               &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;character&gt;
     peak_1     chr1     3670547-3672665      * |       Promoter (&lt;=1kb)
    peak_14     chr1     9298449-9300284      * |       Promoter (&lt;=1kb)
    peak_69     chr1   25228208-25229714      * |       Promoter (&lt;=1kb)
    peak_70     chr1   25504015-25505076      * | Exon (ENSMUST0000014..
    peak_83     chr1   34578812-34580363      * |       Promoter (&lt;=1kb)
        ...      ...                 ...    ... .                    ...
  peak_1902     chr2 178175169-178175887      * |       Promoter (&lt;=1kb)
  peak_1922     chr2 181134082-181135787      * |       Promoter (&lt;=1kb)
  peak_1923     chr2 181156149-181157204      * |       Promoter (&lt;=1kb)
  peak_1925     chr2 181313272-181314836      * |       Promoter (&lt;=1kb)
  peak_1930     chr2 181714476-181715931      * |       Promoter (&lt;=1kb)
            ATAC_geneChr ATAC_geneStart ATAC_geneEnd ATAC_geneLength
               &lt;integer&gt;      &lt;integer&gt;    &lt;integer&gt;       &lt;integer&gt;
     peak_1            1        3214482      3671498          457017
    peak_14            1        8361475      9298573          937099
    peak_69            1       25067476     25228826          161351
    peak_70            1       25276404     25277954            1551
    peak_83            1       34579657     34590944           11288
        ...          ...            ...          ...             ...
  peak_1902            2      178175349    178336491          161143
  peak_1922            2      181075579    181135214           59636
  peak_1923            2      181147653    181157014            9362
  peak_1925            2      181306459    181314500            8042
  peak_1930            2      181715016    181720985            5970
            ATAC_geneStrand ATAC_geneId     ATAC_transcriptId
                  &lt;integer&gt; &lt;character&gt;           &lt;character&gt;
     peak_1               2      497097  ENSMUST00000070533.4
    peak_14               2       71096  ENSMUST00000132064.7
    peak_69               2      210933  ENSMUST00000126626.7
    peak_70               2   100504323  ENSMUST00000193138.1
    peak_83               1      211383 ENSMUST00000052670.10
        ...             ...         ...                   ...
  peak_1902               1       74189  ENSMUST00000108915.7
  peak_1922               2       16536  ENSMUST00000149964.8
  peak_1923               2       13628  ENSMUST00000055990.7
  peak_1925               2       20262  ENSMUST00000103045.3
  peak_1930               1       18389  ENSMUST00000071585.9
            ATAC_distanceToTSS       ATAC_ENSEMBL ATAC_SYMBOL
                     &lt;numeric&gt;        &lt;character&gt; &lt;character&gt;
     peak_1                  0 ENSMUSG00000051951        Xkr4
    peak_14                  0 ENSMUSG00000025909       Sntg1
    peak_69                  0 ENSMUSG00000033569      Adgrb3
    peak_70            -226061 ENSMUSG00000104257     Gm20172
    peak_83                  0 ENSMUSG00000045174       Amer3
        ...                ...                ...         ...
  peak_1902                  0 ENSMUSG00000027525     Phactr3
  peak_1922                  0 ENSMUSG00000016346       Kcnq2
  peak_1923                  0 ENSMUSG00000016349      Eef1a2
  peak_1925                  0 ENSMUSG00000027581       Stmn3
  peak_1930                  0 ENSMUSG00000027584       Oprl1
                     ATAC_GENENAME   ATAC_anno ATAC_logFC ATAC_logCPM    ATAC_F
                       &lt;character&gt; &lt;character&gt;  &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;
     peak_1 X-linked Kx blood gr..    Promoter   0.602984    10.45378  41.78446
    peak_14    syntrophin, gamma 1    Promoter   0.960798     8.76438  46.46418
    peak_69 adhesion G protein-c..    Promoter   0.986539     8.25070  40.79772
    peak_70  predicted gene, 20172        Exon   0.802232     6.45404   8.85783
    peak_83 APC membrane recruit..    Promoter   1.376255     8.69661  86.70410
        ...                    ...         ...        ...         ...       ...
  peak_1902 phosphatase and acti..    Promoter   0.714177     6.72118   8.78683
  peak_1922 potassium voltage-ga..    Promoter   0.912713     8.10930  32.69152
  peak_1923 eukaryotic translati..    Promoter   1.338590     7.82461  54.99971
  peak_1925        stathmin-like 3    Promoter   0.892601     8.12151  30.64904
  peak_1930 opioid receptor-like 1    Promoter   1.213349     7.47748  37.29439
            ATAC_PValue ATAC_qvalue      RNA_Row.names RNA_gene_id  RNA_SYMBOL
              &lt;numeric&gt;   &lt;numeric&gt;        &lt;character&gt; &lt;character&gt; &lt;character&gt;
     peak_1 6.58453e-08 4.81983e-07 ENSMUSG00000051951      497097        Xkr4
    peak_14 1.97965e-08 1.70296e-07 ENSMUSG00000025909       71096       Sntg1
    peak_69 8.55724e-08 6.07481e-07 ENSMUSG00000033569      210933      Adgrb3
    peak_70 4.69864e-03 9.38854e-03 ENSMUSG00000033569      210933      Adgrb3
    peak_83 4.98564e-12 1.57133e-10 ENSMUSG00000045174      211383       Amer3
        ...         ...         ...                ...         ...         ...
  peak_1902 4.85335e-03 9.67494e-03 ENSMUSG00000027525       74189     Phactr3
  peak_1922 8.37488e-07 4.46335e-06 ENSMUSG00000016346       16536       Kcnq2
  peak_1923 2.58482e-09 2.97631e-08 ENSMUSG00000016349       13628      Eef1a2
  peak_1925 1.54857e-06 7.67150e-06 ENSMUSG00000027581       20262       Stmn3
  peak_1930 2.22741e-07 1.40677e-06 ENSMUSG00000027584       18389       Oprl1
                      RNA_GENENAME        RNA_ENSEMBL RNA_geneLength
                       &lt;character&gt;        &lt;character&gt;      &lt;integer&gt;
     peak_1 X-linked Kx blood gr.. ENSMUSG00000051951         457017
    peak_14    syntrophin, gamma 1 ENSMUSG00000025909         938404
    peak_69 adhesion G protein-c.. ENSMUSG00000033569         762232
    peak_70 adhesion G protein-c.. ENSMUSG00000033569         762232
    peak_83 APC membrane recruit.. ENSMUSG00000045174          11288
        ...                    ...                ...            ...
  peak_1902 phosphatase and acti.. ENSMUSG00000027525         219518
  peak_1922 potassium voltage-ga.. ENSMUSG00000016346          59722
  peak_1923 eukaryotic translati.. ENSMUSG00000016349           9362
  peak_1925        stathmin-like 3 ENSMUSG00000027581           8042
  peak_1930 opioid receptor-like 1 ENSMUSG00000027584           5970
            RNA_geneChr RNA_geneStart RNA_geneEnd RNA_geneStrand  RNA_geneId
            &lt;character&gt;     &lt;integer&gt;   &lt;integer&gt;    &lt;character&gt; &lt;character&gt;
     peak_1        chr1       3214482     3671498              -      497097
    peak_14        chr1       8361475     9299878              -       71096
    peak_69        chr1      25067476    25829707              -      210933
    peak_70        chr1      25067476    25829707              -      210933
    peak_83        chr1      34579657    34590944              +      211383
        ...         ...           ...         ...            ...         ...
  peak_1902        chr2     178118975   178338492              +       74189
  peak_1922        chr2     181075579   181135300              -       16536
  peak_1923        chr2     181147653   181157014              -       13628
  peak_1925        chr2     181306459   181314500              -       20262
  peak_1930        chr2     181715016   181720985              +       18389
                 RNA_transcriptId    RNA_anno RNA_logFC RNA_logCPM     RNA_F
                      &lt;character&gt; &lt;character&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
     peak_1  ENSMUST00000070533.4        Gene   1.37498    5.62452   33.7077
    peak_14  ENSMUST00000132064.7        Gene   4.03266    5.98830  192.7039
    peak_69  ENSMUST00000126626.7        Gene   1.96492    8.73183  107.5339
    peak_70  ENSMUST00000126626.7        Gene   1.96492    8.73183  107.5339
    peak_83 ENSMUST00000052670.10        Gene   2.29007    8.17011  191.9172
        ...                   ...         ...       ...        ...       ...
  peak_1902  ENSMUST00000108917.7        Gene   1.75721    7.97049  103.8623
  peak_1922  ENSMUST00000149964.8        Gene   1.08069    9.08113   43.8401
  peak_1923  ENSMUST00000055990.7        Gene   4.06536    6.16441  278.0522
  peak_1925  ENSMUST00000103045.3        Gene   1.87268   12.05944  135.2780
  peak_1930  ENSMUST00000071585.9        Gene   1.87383    9.02372  145.0766
             RNA_PValue  RNA_qvalue     H3K4me3_annotation H3K4me3_geneChr
              &lt;numeric&gt;   &lt;numeric&gt;            &lt;character&gt;       &lt;integer&gt;
     peak_1 2.24273e-03 0.006288707       Promoter (&lt;=1kb)               1
    peak_14 3.88128e-05 0.000618058       Promoter (&lt;=1kb)               1
    peak_69 1.56060e-04 0.001091928       Promoter (&lt;=1kb)               1
    peak_70 1.56060e-04 0.001091928 Exon (ENSMUST0000014..               1
    peak_83 3.90118e-05 0.000618058       Promoter (&lt;=1kb)               1
        ...         ...         ...                    ...             ...
  peak_1902 1.69461e-04 0.001152960       Promoter (&lt;=1kb)               2
  peak_1922 1.25004e-03 0.004047574       Promoter (&lt;=1kb)               2
  peak_1923 1.59710e-05 0.000490337       Promoter (&lt;=1kb)               2
  peak_1925 9.03474e-05 0.000834961       Promoter (&lt;=1kb)               2
  peak_1930 7.64282e-05 0.000816625       Promoter (&lt;=1kb)               2
            H3K4me3_geneStart H3K4me3_geneEnd H3K4me3_geneLength
                    &lt;integer&gt;       &lt;integer&gt;          &lt;integer&gt;
     peak_1           3214482         3671498             457017
    peak_14           8361475         9298573             937099
    peak_69          25067476        25228826             161351
    peak_70          25276404        25277954               1551
    peak_83          34579657        34590944              11288
        ...               ...             ...                ...
  peak_1902         178175349       178336491             161143
  peak_1922         181075579       181135214              59636
  peak_1923         181147653       181157014               9362
  peak_1925         181306459       181314500               8042
  peak_1930         181715016       181720985               5970
            H3K4me3_geneStrand H3K4me3_geneId  H3K4me3_transcriptId
                     &lt;integer&gt;    &lt;character&gt;           &lt;character&gt;
     peak_1                  2         497097  ENSMUST00000070533.4
    peak_14                  2          71096  ENSMUST00000132064.7
    peak_69                  2         210933  ENSMUST00000126626.7
    peak_70                  2      100504323  ENSMUST00000193138.1
    peak_83                  1         211383 ENSMUST00000052670.10
        ...                ...            ...                   ...
  peak_1902                  1          74189  ENSMUST00000108915.7
  peak_1922                  2          16536  ENSMUST00000149964.8
  peak_1923                  2          13628  ENSMUST00000055990.7
  peak_1925                  2          20262  ENSMUST00000103045.3
  peak_1930                  1          18389  ENSMUST00000071585.9
            H3K4me3_distanceToTSS    H3K4me3_ENSEMBL H3K4me3_SYMBOL
                        &lt;numeric&gt;        &lt;character&gt;    &lt;character&gt;
     peak_1                     0 ENSMUSG00000051951           Xkr4
    peak_14                     0 ENSMUSG00000025909          Sntg1
    peak_69                     0 ENSMUSG00000033569         Adgrb3
    peak_70               -226252 ENSMUSG00000104257        Gm20172
    peak_83                     0 ENSMUSG00000045174          Amer3
        ...                   ...                ...            ...
  peak_1902                    56 ENSMUSG00000027525        Phactr3
  peak_1922                     0 ENSMUSG00000016346          Kcnq2
  peak_1923                     0 ENSMUSG00000016349         Eef1a2
  peak_1925                     0 ENSMUSG00000027581          Stmn3
  peak_1930                     0 ENSMUSG00000027584          Oprl1
                  H3K4me3_GENENAME H3K4me3_anno H3K4me3_logFC H3K4me3_logCPM
                       &lt;character&gt;  &lt;character&gt;     &lt;numeric&gt;      &lt;numeric&gt;
     peak_1 X-linked Kx blood gr..     Promoter      0.607114        9.26749
    peak_14    syntrophin, gamma 1     Promoter      0.787977        8.64434
    peak_69 adhesion G protein-c..     Promoter      1.205708        7.52475
    peak_70  predicted gene, 20172         Exon      2.657731        2.48914
    peak_83 APC membrane recruit..     Promoter      1.158232        8.07772
        ...                    ...          ...           ...            ...
  peak_1902 phosphatase and acti..     Promoter      1.655324        3.89074
  peak_1922 potassium voltage-ga..     Promoter      0.750262        8.80515
  peak_1923 eukaryotic translati..     Promoter      1.819209        7.73711
  peak_1925        stathmin-like 3     Promoter      0.615019        8.78460
  peak_1930 opioid receptor-like 1     Promoter      1.978607        7.54088
            H3K4me3_F H3K4me3_PValue H3K4me3_qvalue H3K4me1_annotation
            &lt;numeric&gt;      &lt;numeric&gt;      &lt;numeric&gt;        &lt;character&gt;
     peak_1 135.44293    4.65079e-31    1.28968e-29               &lt;NA&gt;
    peak_14 149.35508    4.76345e-34    1.46952e-32               &lt;NA&gt;
    peak_69 146.70663    1.76407e-33    5.27723e-32               &lt;NA&gt;
    peak_70   6.77423    9.26510e-03    2.58814e-02               &lt;NA&gt;
    peak_83 203.92865    1.03211e-45    4.76122e-44               &lt;NA&gt;
        ...       ...            ...            ...                ...
  peak_1902    14.963    1.10494e-04    4.90024e-04               &lt;NA&gt;
  peak_1922   151.440    1.69973e-34    5.34388e-33               &lt;NA&gt;
  peak_1923   332.565    7.40439e-73    6.76816e-71   Promoter (&lt;=1kb)
  peak_1925   103.048    4.54823e-24    9.80353e-23   Promoter (&lt;=1kb)
  peak_1930   325.324    2.42880e-71    2.10325e-69   Promoter (&lt;=1kb)
            H3K4me1_geneChr H3K4me1_geneStart H3K4me1_geneEnd
                  &lt;integer&gt;         &lt;integer&gt;       &lt;integer&gt;
     peak_1            &lt;NA&gt;              &lt;NA&gt;            &lt;NA&gt;
    peak_14            &lt;NA&gt;              &lt;NA&gt;            &lt;NA&gt;
    peak_69            &lt;NA&gt;              &lt;NA&gt;            &lt;NA&gt;
    peak_70            &lt;NA&gt;              &lt;NA&gt;            &lt;NA&gt;
    peak_83            &lt;NA&gt;              &lt;NA&gt;            &lt;NA&gt;
        ...             ...               ...             ...
  peak_1902            &lt;NA&gt;              &lt;NA&gt;            &lt;NA&gt;
  peak_1922            &lt;NA&gt;              &lt;NA&gt;            &lt;NA&gt;
  peak_1923               2         181147653       181157014
  peak_1925               2         181306459       181314500
  peak_1930               2         181715703       181719372
            H3K4me1_geneLength H3K4me1_geneStrand H3K4me1_geneId
                     &lt;integer&gt;          &lt;integer&gt;    &lt;character&gt;
     peak_1               &lt;NA&gt;               &lt;NA&gt;           &lt;NA&gt;
    peak_14               &lt;NA&gt;               &lt;NA&gt;           &lt;NA&gt;
    peak_69               &lt;NA&gt;               &lt;NA&gt;           &lt;NA&gt;
    peak_70               &lt;NA&gt;               &lt;NA&gt;           &lt;NA&gt;
    peak_83               &lt;NA&gt;               &lt;NA&gt;           &lt;NA&gt;
        ...                ...                ...            ...
  peak_1902               &lt;NA&gt;               &lt;NA&gt;           &lt;NA&gt;
  peak_1922               &lt;NA&gt;               &lt;NA&gt;           &lt;NA&gt;
  peak_1923               9362                  2          13628
  peak_1925               8042                  2          20262
  peak_1930               3670                  1          18389
            H3K4me1_transcriptId H3K4me1_distanceToTSS    H3K4me1_ENSEMBL
                     &lt;character&gt;             &lt;numeric&gt;        &lt;character&gt;
     peak_1                 &lt;NA&gt;                    NA               &lt;NA&gt;
    peak_14                 &lt;NA&gt;                    NA               &lt;NA&gt;
    peak_69                 &lt;NA&gt;                    NA               &lt;NA&gt;
    peak_70                 &lt;NA&gt;                    NA               &lt;NA&gt;
    peak_83                 &lt;NA&gt;                    NA               &lt;NA&gt;
        ...                  ...                   ...                ...
  peak_1902                 &lt;NA&gt;                    NA               &lt;NA&gt;
  peak_1922                 &lt;NA&gt;                    NA               &lt;NA&gt;
  peak_1923 ENSMUST00000055990.7                   -64 ENSMUSG00000016349
  peak_1925 ENSMUST00000103045.3                   911 ENSMUSG00000027581
  peak_1930 ENSMUST00000184127.1                    38 ENSMUSG00000027584
            H3K4me1_SYMBOL       H3K4me1_GENENAME H3K4me1_anno H3K4me1_logFC
               &lt;character&gt;            &lt;character&gt;  &lt;character&gt;     &lt;numeric&gt;
     peak_1           &lt;NA&gt;                   &lt;NA&gt;         &lt;NA&gt;             0
    peak_14           &lt;NA&gt;                   &lt;NA&gt;         &lt;NA&gt;             0
    peak_69           &lt;NA&gt;                   &lt;NA&gt;         &lt;NA&gt;             0
    peak_70           &lt;NA&gt;                   &lt;NA&gt;         &lt;NA&gt;             0
    peak_83           &lt;NA&gt;                   &lt;NA&gt;         &lt;NA&gt;             0
        ...            ...                    ...          ...           ...
  peak_1902           &lt;NA&gt;                   &lt;NA&gt;         &lt;NA&gt;      0.000000
  peak_1922           &lt;NA&gt;                   &lt;NA&gt;         &lt;NA&gt;      0.000000
  peak_1923         Eef1a2 eukaryotic translati..     Promoter      0.932761
  peak_1925          Stmn3        stathmin-like 3     Promoter      0.612051
  peak_1930          Oprl1 opioid receptor-like 1     Promoter      1.086359
            H3K4me1_logCPM H3K4me1_F H3K4me1_PValue H3K4me1_qvalue
                 &lt;numeric&gt; &lt;numeric&gt;      &lt;numeric&gt;      &lt;numeric&gt;
     peak_1             NA        NA             NA             NA
    peak_14             NA        NA             NA             NA
    peak_69             NA        NA             NA             NA
    peak_70             NA        NA             NA             NA
    peak_83             NA        NA             NA             NA
        ...            ...       ...            ...            ...
  peak_1902             NA        NA             NA             NA
  peak_1922             NA        NA             NA             NA
  peak_1923        5.74097   20.7805    5.22563e-06    2.25916e-04
  peak_1925        6.38866   15.0376    1.06225e-04    2.45541e-03
  peak_1930        6.13490   36.7670    1.39101e-09    1.90365e-07
            H3K27me3_annotation H3K27me3_geneChr H3K27me3_geneStart
                    &lt;character&gt;        &lt;integer&gt;          &lt;integer&gt;
     peak_1                &lt;NA&gt;             &lt;NA&gt;               &lt;NA&gt;
    peak_14    Promoter (&lt;=1kb)                1            8361475
    peak_69                &lt;NA&gt;             &lt;NA&gt;               &lt;NA&gt;
    peak_70                &lt;NA&gt;             &lt;NA&gt;               &lt;NA&gt;
    peak_83    Promoter (&lt;=1kb)                1           34579657
        ...                 ...              ...                ...
  peak_1902                &lt;NA&gt;             &lt;NA&gt;               &lt;NA&gt;
  peak_1922    Promoter (&lt;=1kb)                2          181080503
  peak_1923    Promoter (&lt;=1kb)                2          181147653
  peak_1925    Promoter (&lt;=1kb)                2          181306459
  peak_1930                &lt;NA&gt;             &lt;NA&gt;               &lt;NA&gt;
            H3K27me3_geneEnd H3K27me3_geneLength H3K27me3_geneStrand
                   &lt;integer&gt;           &lt;integer&gt;           &lt;integer&gt;
     peak_1             &lt;NA&gt;                &lt;NA&gt;                &lt;NA&gt;
    peak_14          9299238              937764                   2
    peak_69             &lt;NA&gt;                &lt;NA&gt;                &lt;NA&gt;
    peak_70             &lt;NA&gt;                &lt;NA&gt;                &lt;NA&gt;
    peak_83         34590944               11288                   1
        ...              ...                 ...                 ...
  peak_1902             &lt;NA&gt;                &lt;NA&gt;                &lt;NA&gt;
  peak_1922        181134922               54420                   2
  peak_1923        181157014                9362                   2
  peak_1925        181314500                8042                   2
  peak_1930             &lt;NA&gt;                &lt;NA&gt;                &lt;NA&gt;
            H3K27me3_geneId H3K27me3_transcriptId H3K27me3_distanceToTSS
                &lt;character&gt;           &lt;character&gt;              &lt;numeric&gt;
     peak_1            &lt;NA&gt;                  &lt;NA&gt;                     NA
    peak_14           71096  ENSMUST00000140295.7                      0
    peak_69            &lt;NA&gt;                  &lt;NA&gt;                     NA
    peak_70            &lt;NA&gt;                  &lt;NA&gt;                     NA
    peak_83          211383 ENSMUST00000052670.10                   -733
        ...             ...                   ...                    ...
  peak_1902            &lt;NA&gt;                  &lt;NA&gt;                     NA
  peak_1922           16536  ENSMUST00000103049.9                      0
  peak_1923           13628  ENSMUST00000055990.7                      0
  peak_1925           20262  ENSMUST00000103045.3                   -125
  peak_1930            &lt;NA&gt;                  &lt;NA&gt;                     NA
              H3K27me3_ENSEMBL H3K27me3_SYMBOL      H3K27me3_GENENAME
                   &lt;character&gt;     &lt;character&gt;            &lt;character&gt;
     peak_1               &lt;NA&gt;            &lt;NA&gt;                   &lt;NA&gt;
    peak_14 ENSMUSG00000025909           Sntg1    syntrophin, gamma 1
    peak_69               &lt;NA&gt;            &lt;NA&gt;                   &lt;NA&gt;
    peak_70               &lt;NA&gt;            &lt;NA&gt;                   &lt;NA&gt;
    peak_83 ENSMUSG00000045174           Amer3 APC membrane recruit..
        ...                ...             ...                    ...
  peak_1902               &lt;NA&gt;            &lt;NA&gt;                   &lt;NA&gt;
  peak_1922 ENSMUSG00000016346           Kcnq2 potassium voltage-ga..
  peak_1923 ENSMUSG00000016349          Eef1a2 eukaryotic translati..
  peak_1925 ENSMUSG00000027581           Stmn3        stathmin-like 3
  peak_1930               &lt;NA&gt;            &lt;NA&gt;                   &lt;NA&gt;
            H3K27me3_anno H3K27me3_logFC H3K27me3_logCPM H3K27me3_F
              &lt;character&gt;      &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;
     peak_1          &lt;NA&gt;       0.000000              NA         NA
    peak_14      Promoter      -0.578443         7.07069    8.17212
    peak_69          &lt;NA&gt;       0.000000              NA         NA
    peak_70          &lt;NA&gt;       0.000000              NA         NA
    peak_83      Promoter      -0.926854         5.73161    7.12230
        ...           ...            ...             ...        ...
  peak_1902          &lt;NA&gt;        0.00000              NA         NA
  peak_1922      Promoter       -1.01484         9.08503    81.5584
  peak_1923      Promoter       -1.18230         8.57170    84.1744
  peak_1925      Promoter       -1.14450         6.24934    16.4223
  peak_1930          &lt;NA&gt;        0.00000              NA         NA
            H3K27me3_PValue H3K27me3_qvalue     H3K27ac_annotation
                  &lt;numeric&gt;       &lt;numeric&gt;            &lt;character&gt;
     peak_1              NA              NA       Promoter (&lt;=1kb)
    peak_14      0.00426473       0.0616407       Promoter (&lt;=1kb)
    peak_69              NA              NA       Promoter (&lt;=1kb)
    peak_70              NA              NA Exon (ENSMUST0000014..
    peak_83      0.00762819       0.0899976       Promoter (&lt;=1kb)
        ...             ...             ...                    ...
  peak_1902              NA              NA       Promoter (&lt;=1kb)
  peak_1922     2.09500e-19     7.17349e-17       Promoter (&lt;=1kb)
  peak_1923     5.65224e-20     2.06724e-17       Promoter (&lt;=1kb)
  peak_1925     5.11572e-05     1.99691e-03       Promoter (&lt;=1kb)
  peak_1930              NA              NA       Promoter (&lt;=1kb)
            H3K27ac_geneChr H3K27ac_geneStart H3K27ac_geneEnd
                  &lt;integer&gt;         &lt;integer&gt;       &lt;integer&gt;
     peak_1               1           3214482         3671498
    peak_14               1           8361475         9298573
    peak_69               1          25067476        25228826
    peak_70               1          25276404        25277954
    peak_83               1          34579657        34590944
        ...             ...               ...             ...
  peak_1902               2         178175349       178336491
  peak_1922               2         181080503       181134922
  peak_1923               2         181147653       181157014
  peak_1925               2         181306459       181314500
  peak_1930               2         181715016       181720985
            H3K27ac_geneLength H3K27ac_geneStrand H3K27ac_geneId
                     &lt;integer&gt;          &lt;integer&gt;    &lt;character&gt;
     peak_1             457017                  2         497097
    peak_14             937099                  2          71096
    peak_69             161351                  2         210933
    peak_70               1551                  2      100504323
    peak_83              11288                  1         211383
        ...                ...                ...            ...
  peak_1902             161143                  1          74189
  peak_1922              54420                  2          16536
  peak_1923               9362                  2          13628
  peak_1925               8042                  2          20262
  peak_1930               5970                  1          18389
             H3K27ac_transcriptId H3K27ac_distanceToTSS    H3K27ac_ENSEMBL
                      &lt;character&gt;             &lt;numeric&gt;        &lt;character&gt;
     peak_1  ENSMUST00000070533.4                     0 ENSMUSG00000051951
    peak_14  ENSMUST00000132064.7                     0 ENSMUSG00000025909
    peak_69  ENSMUST00000126626.7                     0 ENSMUSG00000033569
    peak_70  ENSMUST00000193138.1               -226214 ENSMUSG00000104257
    peak_83 ENSMUST00000052670.10                     0 ENSMUSG00000045174
        ...                   ...                   ...                ...
  peak_1902  ENSMUST00000108915.7                     0 ENSMUSG00000027525
  peak_1922  ENSMUST00000103049.9                     0 ENSMUSG00000016346
  peak_1923  ENSMUST00000055990.7                     0 ENSMUSG00000016349
  peak_1925  ENSMUST00000103045.3                     0 ENSMUSG00000027581
  peak_1930  ENSMUST00000071585.9                     0 ENSMUSG00000027584
            H3K27ac_SYMBOL       H3K27ac_GENENAME H3K27ac_anno H3K27ac_logFC
               &lt;character&gt;            &lt;character&gt;  &lt;character&gt;     &lt;numeric&gt;
     peak_1           Xkr4 X-linked Kx blood gr..     Promoter       1.25269
    peak_14          Sntg1    syntrophin, gamma 1     Promoter       2.25132
    peak_69         Adgrb3 adhesion G protein-c..     Promoter       1.71288
    peak_70        Gm20172  predicted gene, 20172         Exon       2.45135
    peak_83          Amer3 APC membrane recruit..     Promoter       1.44331
        ...            ...                    ...          ...           ...
  peak_1902        Phactr3 phosphatase and acti..     Promoter      1.382198
  peak_1922          Kcnq2 potassium voltage-ga..     Promoter      0.909119
  peak_1923         Eef1a2 eukaryotic translati..     Promoter      1.426917
  peak_1925          Stmn3        stathmin-like 3     Promoter      1.337649
  peak_1930          Oprl1 opioid receptor-like 1     Promoter      1.363849
            H3K27ac_logCPM H3K27ac_F H3K27ac_PValue H3K27ac_qvalue        name
                 &lt;numeric&gt; &lt;numeric&gt;      &lt;numeric&gt;      &lt;numeric&gt; &lt;character&gt;
     peak_1        8.81735  151.4959    1.67395e-34    4.08130e-32      peak_1
    peak_14        7.44136  188.6314    1.91288e-42    6.62156e-40     peak_14
    peak_69        7.75798  148.3754    7.82086e-34    1.80483e-31     peak_69
    peak_70        5.04358   34.5321    4.36085e-09    1.34796e-07     peak_70
    peak_83        7.35920   86.0818    2.18441e-20    2.14293e-18     peak_83
        ...            ...       ...            ...            ...         ...
  peak_1902        6.22622   35.9816    2.07855e-09    6.74235e-08   peak_1902
  peak_1922        8.31643   64.4153    1.15104e-15    7.72601e-14   peak_1922
  peak_1923        7.39512   86.2073    2.05151e-20    2.03730e-18   peak_1923
  peak_1925        8.00741  111.4177    7.08864e-26    1.02853e-23   peak_1925
  peak_1930        7.84616  104.9298    1.79151e-24    2.34334e-22   peak_1930
                    ATAC_RNA   annotated_genes    dist_to_TSS
                 &lt;character&gt;   &lt;CharacterList&gt;  &lt;IntegerList&gt;
     peak_1 Active Promoters              Xkr4              0
    peak_14 Active Promoters             Sntg1              0
    peak_69 Active Promoters    Lmbrd1,Gm20172   549578,48240
    peak_70 Active Promoters    Gm20172,Adgrb3 -226061,324631
    peak_83 Active Promoters             Amer3              0
        ...              ...               ...            ...
  peak_1902 Active Promoters     Phactr3,Sycp2   56194,231798
  peak_1922 Active Promoters      Chrna4,Kcnq2       -90536,0
  peak_1923 Active Promoters            Eef1a2              0
  peak_1925 Active Promoters Gmeb2,Stmn3,Rtel1 -25237,0,-4903
  peak_1930 Active Promoters             Oprl1              0
  -------
  seqinfo: 21 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/shinyReport-dispatch.html">shinyReport</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getEnrichmentTable-dispatch.html">getEnrichmentTable</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="va">sig_go_ids</span> <span class="op">=</span> <span class="va">tb</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="va">tb</span><span class="op">$</span><span class="va">p_adjust</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span></span>
<span><span class="va">cl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/simplifyEnrichment/man/simplifyGO.html">simplifyGO</a></span><span class="op">(</span>mat <span class="op">=</span> <span class="va">sig_go_ids</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>You haven't provided value for `ont`, guess it as `BP`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>relations: is_a, part_of, regulates, negatively_regulates, positively_regulates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>IC_method: IC_annotation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>term_sim_method: Sim_XGraSM_2013</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>IC_method: IC_annotation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Cluster 33 terms by 'binary_cut'... 4 clusters, used 0.09793186 secs.
'magick' package is suggested to install to give better rasterization.

Set `ht_opt$message = FALSE` to turn off this message.
Perform keywords enrichment for 4 GO lists...</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="day2-4_enrichment_analysis_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<section id="output" class="level3"><h3 class="anchored" data-anchor-id="output">Output</h3>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><p>The heatmap shows clear clustering of GO terms into two main clusters, indicating distinct functional groups.</p></li>
<li><p>The GO terms on the right highlight biological processes relevant to neuronal and ion transport functions, such as transport, inorganic cation ion, exocytosis, synaptic, neurotransmitter, and behavior learning.</p></li>
<li><p>The cluster with terms like cytosolic concentration, calcium ion suggests a calcium-related signaling or ion homeostasis process at play.</p></li>
<li><p>The color gradient and similarity scores (up to ~0.6) indicate moderate similarity within clusters, which is reasonable for GO term groupings.</p></li>
</ul>
</div>
</div>
</section><section id="interpretation" class="level3"><h3 class="anchored" data-anchor-id="interpretation">Interpretation</h3>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><p>E11.5 to E15.5 is a key window in neural development involving neuronal differentiation, synapse formation, and active signaling.</p></li>
<li><p>Active promoters linked to ion transport, synaptic transmission, calcium signaling are expected to be regulated during this developmental stage.</p></li>
<li><p>The clustering separating signaling and transport-related terms from behavior and learning terms also aligns with early neural circuit formation and functional maturation.</p></li>
</ul>
</div>
</div>
</section></section><section id="question" class="level2"><h2 class="anchored" data-anchor-id="question">Question</h2>
<p><strong>Perform enrichment analysis of any other category.</strong></p>


</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "î§‹";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/sib-swiss\.github\.io\/biology-informed-multiomics\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->



</body></html>