---
title: "Exercise 5"
editor: source
editor_options: 
  chunk_output_type: console
---

# Integrating ATAC and RNA sequencing data

## Learning Objectives


## Load Libraries
```{r setup, include=FALSE} 
knitr::opts_chunk$set(warning = FALSE) 
ht_opt$message = FALSE
```

```{r}
#| warning: false
#| message: false
library(ComplexHeatmap)
library(EnrichedHeatmap)
library(circlize)
library(SummarizedExperiment)
```


## Load required objects from previous exercises
Here we load the `SummarizedExperiment` for the ChIP-seq data
```{r}
atac <- readRDS("data/atac_se.rds")
rna <- readRDS("data/rna_se.rds")
rownames(colData(rna)) <- gsub(pattern = "\\.tsv.gz", replacement = "", x = rownames(colData(rna)))

overlap <- readRDS("output/overlap_data.rds")

mat_ATAC <- readRDS("./output/mat_atac.rds")
mat_RNA <- readRDS("./output/mat_rna.rds")
```


## Add possible genomic regions for ATAC and RNA
```{r}
overlap$ATAC_RNA <- NULL

# Increased accessibility and increased expression
overlap$ATAC_RNA[overlap$ATAC_logFC > 0 & overlap$RNA_logFC > 0] <- "Active"

# Increased accessibility and decreased expression
overlap$ATAC_RNA[overlap$ATAC_logFC > 0 & overlap$RNA_logFC < 0] <- "Repressed"

# Decreased accessibility and decreased expression
overlap$ATAC_RNA[overlap$ATAC_logFC < 0 & overlap$RNA_logFC < 0] <- "Silent"

# Decreased accessibility and increased expression (incongruent)
overlap$ATAC_RNA[overlap$ATAC_logFC < 0 & overlap$RNA_logFC > 0] <- "Incongruent"

# Accessibility changes, but no gene expression info
overlap$ATAC_RNA[overlap$ATAC_logFC < 0 & is.na(overlap$RNA_logFC)] <- "Dec. Acc."
overlap$ATAC_RNA[overlap$ATAC_logFC > 0 & is.na(overlap$RNA_logFC)] <- "Inc. Acc."

saveRDS(object = overlap, file = "output/overlap_anno.rds")
```


## `EnrichedHeatmap` + `ComplexHeatmap`

### Our function from previous exercise
```{r}
make_EH <- function(norm_mat, heatmap_cols = c("white", "red"), split_rows = NULL, hm_name, col_fill = "#ffcccc"){
  col_fun <- colorRamp2(quantile(norm_mat, c(0.01, 0.99)), heatmap_cols)  
  
  vmin <- as.numeric(quantile(norm_mat, c(0.01)))
  vmax <- as.numeric(quantile(norm_mat, c(0.99)))
  vmid <- (vmin + vmax) / 2
  legend_ticks <- c(vmin, vmid, vmax)

EnrichedHeatmap(
  mat = norm_mat,
  name = hm_name,
  row_split = split_rows,
  col = col_fun,
  width = unit(2, "cm"),
  height = unit(8, "cm"),
  column_title = hm_name,
  column_title_gp = gpar(fontsize = 8, fill = col_fill),
  axis_name = c("-1kb", "mid", "1kb"),
  heatmap_legend_param = list(
    at = legend_ticks,
    legend_height = unit(0.5, "cm"),
    legend_width = unit(0.1, "cm"),
    labels = round(legend_ticks, digits = 1),
    title_gp = gpar(fontsize = 8),
    labels_gp = gpar(fontsize = 7)
  ),
  top_annotation = HeatmapAnnotation(
    lines = anno_enriched(
      height = unit(1, "cm"),
      axis_param = list(
        side = "right",
        facing = "inside",
        gp = gpar(
          fontsize = 7,
          lwd = 0.4
        )
      )
    )
  )
)
}
```


### ATAC
```{r}
split_anno <- overlap$ATAC_RNA
names(split_anno) <- names(overlap)

head(split_anno)

cols_an <- RColorBrewer::brewer.pal(n = length(unique(split_anno)), name = "Set1")

eh_11h <- make_EH(norm_mat = mat_ATAC$ATAC_11half, hm_name = "AS-E11.5", col_fill = "#ffcccc")
eh_15h <- make_EH(norm_mat = mat_ATAC$ATAC_15half, hm_name = "AS-E15.5", col_fill = "#e6fff2")

row_order_eh <- row_order(eh_11h)

anno_hm <- Heatmap(
  mat = split_anno,
  col = cols_an, 
  name = "Annotation",
  show_row_names = FALSE, 
  show_column_names = FALSE, 
  width = unit(2, "mm"),
  height = unit(8, "cm"),
  row_order = row_order_eh,
  row_title_gp = gpar(fontsize = 0)
)

draw(anno_hm + eh_11h + eh_15h, split = split_anno, merge_legend = TRUE)
```


### RNA
```{r}
rna_hm <- Heatmap(matrix = mat_RNA, 
        name = "RNA", 
        cluster_columns = FALSE, 
        cluster_rows = FALSE,
        na_col = "grey",
        row_order = row_order_eh,
        show_row_names = FALSE, 
        show_column_names = FALSE,
        row_title_rot = 0,
        top_annotation = HeatmapAnnotation(
          df = colData(rna)[,2,drop = FALSE], 
          annotation_name_gp = gpar(fontsize = 0)
        ),
        width = unit(2, "cm"),
        height = unit(8, "cm"),
        heatmap_legend_param = list(
          legend_height = unit(0.5, "cm"),
          legend_width = unit(0.1, "cm"),
          at = c(-10,0,10), 
          title = "RNA",
          title_gp = gpar(fontsize = 8),
          labels_gp = gpar(fontsize = 7)
        )
)

draw(rna_hm, split = split_anno, merge_legend = TRUE)
```


### Combine plots
```{r}
draw(anno_hm + eh_11h + eh_15h + rna_hm, split = split_anno, merge_legend = TRUE)
```


## Question
**Can you change order of the regions?**
1. Active
2. Repressed
3. Silent
4. Incongruent
5. Dec. Acc.
6. Inc. Acc.

_**Hint:**_ You would have to change the factor levels of `split_anno`.

:::{.callout-tip collapse="true"}

### Answer

```{r}
unique(split_anno)
split_anno <- factor(split_anno, levels = unique(split_anno)[c(1,6,5,4,2,3)])

draw(anno_hm + eh_11h + eh_15h + rna_hm, split = split_anno, merge_legend = FALSE)
```

:::

:::{.callout-important}
To make the plots a bit prettier, you can run these codes before makeing the plot. You only need to run these once while you are in an `R` session, not before making every plot.

```{r}
ht_opt$TITLE_PADDING <- unit(1, "mm")
ht_opt$legend_gap <- unit(3, "mm")
ht_opt$legend_grid_height <- unit(2, "mm")
ht_opt$legend_grid_width <- unit(2, "mm")
ht_opt$HEATMAP_LEGEND_PADDING <- unit(1, "mm")
ht_opt$heatmap_border <- TRUE
```

:::
